### PR TITLE
fix: [select-dialog] engrampa show select-dialog crash in wayland

### PR DESCRIPTION
**Pull request by GongHeng2017**
1. the show top logic error
2. adjust the show top logic

Log: fix issue
Bug: https://pms.uniontech.com/bug-view-202677.html

### CODE REVIEW

File: src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp [hunk 1]
```diff
@@ -489,6 +489,12 @@ void FileDialogHandle::show()
             d->dialog->statusBar()->setMode(FileDialogStatusBar::Mode::kOpen);
             d->dialog->updateAsDefaultSize();
             d->dialog->moveCenter();
+            QVariant isGtk = qApp->property("GTK");
```
Comment (Clauszy) [inline src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp]: 请封装成函数

**Review by Clauszy — CHANGES_REQUESTED** [review]

File: src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp [hunk 2]
```diff
@@ -489,6 +489,12 @@ void FileDialogHandle::show()
...
+            if (WindowUtils::isWayLand() && isGtk.isValid() && isGtk.toBool()) {
+                QFunctionPointer setWindowProperty = qApp->platformFunction("_d_setWindowProperty");
```
Comment (Clauszy) [inline src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp]: 没做指针检查

**Review by Clauszy — CHANGES_REQUESTED** [review]

File: src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp [hunk 3]
```diff
@@ -489,6 +489,12 @@ void FileDialogHandle::show()
...
+                // set window stay on top
+                reinterpret_cast<void(*)(QWindow *, const char *, const QVariant &)>(setWindowProperty)(d->dialog->windowHandle(), "_d_dwayland_staysontop", true);
```
Comment (Kakueeen) [inline src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp]: setWindowProperty不需要判空吗

**Review by Kakueeen — APPROVED** [review]

File: src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp (hunk 3 above)
Comment (GongHeng2017) [reply src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp]: 需要

File: src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp (hunk 2 above)
Comment (GongHeng2017) [reply src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp]: 需要

File: src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp (hunk 1 above)
Comment (GongHeng2017) [reply src/plugins/filedialog/filedialogplugin-core/dbus/filedialoghandle.cpp]: ok

**Review by Clauszy — APPROVED** [review]

**Review by Kakueeen — APPROVED** [review]

**Pull request merged**